extensions/ql-vscode/src/remote-queries/remote-query-result-index.ts

```diff
@@ -3,11 +3,20 @@ export interface RemoteQueryResultIndex {
   items: RemoteQueryResultIndexItem[];
 }
 
-export interface RemoteQueryResultIndexItem {
+export interface RemoteQuerySuccessIndexItem {
   id: string;
   artifactId: number;
   nwo: string;
   resultCount: number;
   bqrsFileSize: number;
   sarifFileSize?: number;
 }
+
+export interface RemoteQueryFailureIndexItem {
+  id: string;
+  artifactId: number;
+  nwo: string;
+  error: string;
+}
+
+export type RemoteQueryResultIndexItem = RemoteQuerySuccessIndexItem & RemoteQueryFailureIndexItem;
```

Comment:
I think this is the correct type to mean that `RemoteQueryResultIndexItem` is either a `RemoteQuerySuccessIndexItem` _or_ a RemoteQueryFailureIndexItem`.

```suggestion
export type RemoteQueryResultIndexItem = RemoteQuerySuccessIndexItem | RemoteQueryFailureIndexItem;
```